### ISSUE TITLE
Flaky test: Testing Incorrect locale

### ISSUE THREAD

**Issue by garethbowen** [medic/cht-core#8788]
**Describe the issue**

This test has failed three times in the last three days with two different stacktraces:

Builds:

- https://github.com/medic/cht-core/actions/runs/7444119349/job/20250181164#step:17:786
- https://github.com/medic/cht-core/actions/runs/7418970237/job/20187961908
- https://github.com/medic/cht-core/actions/runs/7417779409/job/20184882776

First error:

```
------------------------------------------------------------------
[chrome-headless-shell 120.0.6099.109 linux #0-5] Running: chrome-headless-shell (v120.0.6099.109) on linux
[chrome-headless-shell 120.0.6099.109 linux #0-5] Session ID: af35b35aa661933827cb13dfe74e9267
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] » /tests/e2e/default/translations/incorrect-locale.wdio-spec.js
[chrome-headless-shell 120.0.6099.109 linux #0-5] Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ "before all" hook for Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ "after all" hook for Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] 2 failing (5m 5.2s)
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] 1) Testing Incorrect locale "before all" hook for Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5] Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "before all" hook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 120.0.6099.109 linux #0-5] Error: Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "before all" hook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at createTimeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/errors.js:498:15)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at Hook.Runnable._timeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:431:10)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at Timeout.<anonymous> (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:246:24)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at listOnTimeout (node:internal/timers:559:17)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at processTimers (node:internal/timers:502:7)
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] 2) Testing Incorrect locale "after all" hook for Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5] Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "after all" hook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 120.0.6099.109 linux #0-5] Error: Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "after all" hook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at createTimeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/errors.js:498:15)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at Hook.Runnable._timeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:431:10)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at Timeout.<anonymous> (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:246:24)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at listOnTimeout (node:internal/timers:559:17)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at processTimers (node:internal/timers:502:7)
------------------------------------------------------------------
```

Second error:

```
------------------------------------------------------------------
[chrome-headless-shell 120.0.6099.109 linux #0-5] Running: chrome-headless-shell (v120.0.6099.109) on linux
[chrome-headless-shell 120.0.6099.109 linux #0-5] Session ID: fa7b6a7a753c09806ac2c849d5e97154
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] » /tests/e2e/default/translations/incorrect-locale.wdio-spec.js
[chrome-headless-shell 120.0.6099.109 linux #0-5] Testing Incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]    ✖ should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] 1 failing (4m 7.5s)
[chrome-headless-shell 120.0.6099.109 linux #0-5]
[chrome-headless-shell 120.0.6099.109 linux #0-5] 1) Testing Incorrect locale should work with incorrect locale
[chrome-headless-shell 120.0.6099.109 linux #0-5] element ("#header-dropdown-link") still not displayed after 40000ms
[chrome-headless-shell 120.0.6099.109 linux #0-5] Error: element ("#header-dropdown-link") still not displayed after 40000ms
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at async waitForAngularLoaded (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:302:3)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at async Object.waitForPageLoaded (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:308:3)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at async Object.setLanguage (/home/runner/work/cht-core/cht-core/tests/page-objects/default/users/user-settings.wdio.page.js:23:3)
[chrome-headless-shell 120.0.6099.109 linux #0-5]     at async Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js:49:5)
```

**Describe alternatives you've considered**

This test was [worked on recently](https://github.com/medic/cht-core/issues/8535) but the stacktrace is different.

**Comment by garethbowen** [comment]
Another failure: https://github.com/medic/cht-core/actions/runs/8755366587/job/24029891368

**Comment by garethbowen** [comment]
Attempted fix merged. Reopen if it happens again..

**Comment by garethbowen** [comment]
Still flaking...

https://github.com/medic/cht-core/actions/runs/8919119214/job/24495043656?pr=9083

```
[chrome-headless-shell 124.0.6367.60 linux #0-5] Running: chrome-headless-shell (v124.0.6367.60) on linux
[chrome-headless-shell 124.0.6367.60 linux #0-5] Session ID: 09bf7dbe7e09605be71d7310334d4274
[chrome-headless-shell 124.0.6367.60 linux #0-5]
[chrome-headless-shell 124.0.6367.60 linux #0-5] » /tests/e2e/default/translations/incorrect-locale.wdio-spec.js
[chrome-headless-shell 124.0.6367.60 linux #0-5] Testing Incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5]    ? should work with incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5]    ✖ "before each" hook for Testing Incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5]    ✖ "after each" hook for Testing Incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5]
[chrome-headless-shell 124.0.6367.60 linux #0-5] 2 failing (5m 4.5s)
[chrome-headless-shell 124.0.6367.60 linux #0-5]
[chrome-headless-shell 124.0.6367.60 linux #0-5] 1) Testing Incorrect locale "before each" hook for Testing Incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5] Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "before each" hook: wrappedHook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 124.0.6367.60 linux #0-5] Error: Timeout of 120000ms exceeded. The execution in the test "Testing Incorrect locale "before each" hook: wrappedHook for "should work with incorrect locale"" took too long. Try to reduce the run time or increase your timeout for test specs (https://webdriver.io/docs/timeouts). (/home/runner/work/cht-core/cht-core/tests/e2e/default/translations/incorrect-locale.wdio-spec.js)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at createTimeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/errors.js:498:15)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at Hook.Runnable._timeoutError (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:431:10)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at Timeout.<anonymous> (/home/runner/work/cht-core/cht-core/node_modules/mocha/lib/runnable.js:246:24)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at listOnTimeout (node:internal/timers:573:17)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at processTimers (node:internal/timers:514:7)
[chrome-headless-shell 124.0.6367.60 linux #0-5]
[chrome-headless-shell 124.0.6367.60 linux #0-5] 2) Testing Incorrect locale "after each" hook for Testing Incorrect locale
[chrome-headless-shell 124.0.6367.60 linux #0-5] Timeout
[chrome-headless-shell 124.0.6367.60 linux #0-5] Error: Timeout
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at listOnTimeout (node:internal/timers:573:17)
[chrome-headless-shell 124.0.6367.60 linux #0-5]     at processTimers (node:internal/timers:514:7)
```

**Comment by Benmuiruri** [comment]
Hi @garethbowen is this test fix stable enough to be added to 4.8.0 ?

**Comment by garethbowen** [comment]
@Benmuiruri Thanks for finding this. I've added it to 4.8.0 for now.
